core/templates/dev/head/pages/exploration-player-page/services/question-player-engine.service.ts

```diff
@@ -35,18 +36,32 @@ oppia.factory('QuestionPlayerEngineService', [
   '$http', '$q', '$rootScope', 'AlertsService', 'AnswerClassificationService',
   'ContextService', 'ExplorationHtmlFormatterService',
   'ExpressionInterpolationService', 'FocusManagerService',
-  'QuestionObjectFactory', 'StateCardObjectFactory',
-  'INTERACTION_DISPLAY_MODE_INLINE', 'INTERACTION_SPECS',
+  'QuestionObjectFactory', 'ReadOnlyExplorationBackendApiService',
+  'StateCardObjectFactory', 'UrlService', 'INTERACTION_DISPLAY_MODE_INLINE',
+  'INTERACTION_SPECS',
   function(
       $http, $q, $rootScope, AlertsService, AnswerClassificationService,
       ContextService, ExplorationHtmlFormatterService,
       ExpressionInterpolationService, FocusManagerService,
-      QuestionObjectFactory, StateCardObjectFactory,
-      INTERACTION_DISPLAY_MODE_INLINE, INTERACTION_SPECS) {
+      QuestionObjectFactory, ReadOnlyExplorationBackendApiService,
+      StateCardObjectFactory, UrlService, INTERACTION_DISPLAY_MODE_INLINE,
+      INTERACTION_SPECS) {
     var _explorationId = ContextService.getExplorationId();
-
-    var version = GLOBALS.explorationVersion;
-
+    var version = UrlService.getExplorationVersionFromUrl();
+
+    if (version) {
+      ReadOnlyExplorationBackendApiService
+        .loadExploration(_explorationId, version)
+        .then(function (exploration) {
+          version = exploration.version;
+        });
+    } else {
+      ReadOnlyExplorationBackendApiService
+        .loadExploration(_explorationId, version)
+        .then(function (exploration) {
+          version = exploration.version;
+        });
+    }
```

Comment:
Why are these two the same?

Reply:
Changed.
The second was supposed to load the exploration without the version (latest exploration).